### PR TITLE
Startup Check

### PR DESCRIPTION
**Pull request by nguyenfilip**
Candlepin will quickly fail at startup if the Qpid Broker is not running and healthy.

### CODE REVIEW

**Comment by nguyenfilip** [comment]
retest this please

File: server/src/main/java/org/candlepin/audit/QpidQmf.java [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+        log.debug("Getting info about queue {}", queueName);
+        List<Map<String, Object>> mm =  runQuery("_object_id", singularMap("_object_name", queueName));
+        if (mm.size() == 0) {
+            throw new RuntimeException("Couldn't find a queue in Qpid: " + queueName);
+        }
+        if (mm.size() > 1) {
+            throw new RuntimeException("Found unexpected amount of information about queue: " + queueName);
+        }
+        return mm.get(0);
+    }
```
Comment (mstead) [inline server/src/main/java/org/candlepin/audit/QpidQmf.java]: This should be private unless there are plans to use it outside the class.

File: server/src/main/java/org/candlepin/audit/QpidQmf.java [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+    /**
+     * Map with one entry, key and value
+     * @param key
+     * @param value
+     * @return Map with one entry
+     */
+    public static Map<Object, Object> singularMap(Object key, Object value) {
```
Comment (mstead) [inline server/src/main/java/org/candlepin/audit/QpidQmf.java]: This should be private unless there are plans to use it outside the class.

File: server/src/main/java/org/candlepin/audit/QpidQmf.java [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+    }
+
+    /**
+     * Finds fully qualified names of queues that are bound to an exchange
+     *
+     * @param exchangeName
+     * @return Fully qualified names of queues
+     * @throws JMSException Error connecting
+     */
+    public Set<String> getExchangeBoundQueueNames(String exchangeName) throws JMSException {
```
Comment (nguyenfilip) [inline server/src/main/java/org/candlepin/audit/QpidQmf.java]: ack

File: server/src/main/java/org/candlepin/audit/QpidQmf.java (hunk 1 above)
Comment (nguyenfilip) [inline server/src/main/java/org/candlepin/audit/QpidQmf.java]: ack

File: server/src/main/java/org/candlepin/audit/QpidQmf.java (hunk 2 above)
Comment (nguyenfilip) [inline server/src/main/java/org/candlepin/audit/QpidQmf.java]: ack

**Pull request merged**